src/advanced-types/type-bounds.md

```diff
@@ -1 +1,55 @@
 # Type Bounds
+When we express a [subtype-supertype relationship](/advanced-types/subtyping.html) by writing `T <: U`, then we say that `T` is *bounded* by `U`. Lets explore *type bounds* in more detail.
+
+## The `Any` and `None` types
+All types in Motoko are bounded by a special type, namely the `Any` type. This type is the *supertype* of all types and thus all types are a *subtype* of the `Any` type. We may refer to it as the *top type*. A variable in Motoko can be of type `Any`.
+
+Another special type in Motoko is the `None` type. This type is the *subtype* of all types and thus all types are a *supertype* of `None`. No value in Motoko can have the `None` type. 
+
+For any type `T` in Motoko, the following *type bounds* hold:  
+```motoko
+None <: T
+
+T <: Any
+```
+
+## Type bounds in functions
+Consider the following *function signature*:
+```motoko
+func makeNat<T <: Number>(x : T) : Natural
```

Comment:
```suggestion
func makeNat<T <: Number>(x : T) : Natural
```
Actually, this isn't a great example, since you can just write it as 

```
func makeNat(x : Number) : Natural
```
The bound isn't necessary.

Something along the lines of:
```
func compare<T <: { compare: T ->  Order.Order }>(T o1,T o2 ) : Bool { o1.compare(o2) }
```
might be better. Here we use the bound to access a comparison function on `o1` to compare it against `o2`.

There might be better examples out there for other languages, worth searching.

Reply:
https://en.wikipedia.org/wiki/Bounded_quantification might be a good starting place.